app/models/rubygem.rb

```diff
@@ -177,7 +177,9 @@ def to_param
   end
 
   def pushable?
-    new_record? || versions.indexed.count.zero?
+    return true if new_record?
+    return false if versions.indexed.count.nonzero?
+    updated_at < 100.days.ago || created_at > 30.days.ago
```

Comment:
Although, I agree that:
```Ruby
return true if new_record?
return false if versions.indexed.count.nonzero?
updated_at < 100.days.ago || created_at > 30.days.ago
```
is not very elegant, using:
```Ruby
new_record? || versions.indexed.none? || !protected?
```
would make every rubygem older than a month and updated in last 100 days pushable. `protected?` function could look like:
```Ruby
# gem with no indexed versions and updated(yanked) in last 100 days and older than a month
versions.indexed.none? && updated_at > 100.days.ago && created_at < 30.days.ago
```
I didn't want to write a long boolean query to begin with, hence those guard statements.
May be there could be a functions which just checks for date range?

Let me know what do you think :smiley: